txnsync/bloomFilter.go

```diff
@@ -142,6 +146,10 @@ func (s *syncState) makeBloomFilter(encodingParams requestParams, txnGroups []po
 		// we want none.
 		return
 	}
+	minGroupCounter := uint64(0)
+	if hintPrevBloomFilter != nil {
+		minGroupCounter = hintPrevBloomFilter.containedTxnsRange.lastCounter
```

Comment:
The `hintPrevBloomFilter` is a previously generated bloom filter.. but it's not the last one sent to the peer.

Reply:
also, it would be incorrect to use the lastCounter from the latest bloom filter that *was* sent to the peer, unless the  encodingParams are identical. ( that's what I addressed on my branch.. )